modules/costs/app/components/time_entries/days_and_hours_form.rb

```diff
@@ -80,6 +80,26 @@ class DaysAndHoursForm < ApplicationForm
 
     private
 
+    def start_time_in_local_time
+      return if model.start_timestamp.blank?
+
+      if model.user == User.current
+        model.start_timestamp.in_time_zone(User.current.time_zone)
+      else
+        model.start_timestamp
+      end.strftime("%H:%M")
```

Comment:
Shouldn't it always be `model.start_timestamp.in_time_zone(model.user.time_zone)`?

That may be the reason why the times are not correctly shown when editing the time entries for someone else.